### PR TITLE
enable system controller config

### PR DESCRIPTION
**Pull request by changweige**
1. move system controller config to a dedicated struct
2. capable of configuring daemon cpu profile duration

Signed-off-by: Changwei Ge <gechangwei@bytedance.com>

### CODE REVIEW

File: misc/snapshotter/config.toml [hunk 1]
```diff
@@ -4,13 +4,18 @@ root = "/var/lib/containerd-nydus"
...
 enable_stargz = false
 # Whether snapshotter should try to clean up resources when it is closed
 cleanup_on_close = false
 
+[system]
+# Snapshotter's debug and trace HTTP server interface
+enable = true
+
+[system.debug]
+daemon_cpu_profile_duration_secs = 5
```
Comment (sctb512) [inline misc/snapshotter/config.toml:17]: Shall we add a comment for this item in `toml` file?

File: snapshot/snapshot.go [hunk 2]
```diff
@@ -122,7 +122,7 @@ func NewSnapshotter(ctx context.Context, cfg *config.SnapshotterConfig) (snapsho
 		}()
 	}
 
-	if cfg.EnableSystemController {
+	if config.SystemControllerEnabled() {
```
Comment (sctb512) [inline snapshot/snapshot.go:125]: Maybe `IsSystemControllerEnabled` or `GetSystemControllerEnabled` is better to understand?

**Review by sctb512 — COMMENTED** [review]
Others, LGTM

File: misc/snapshotter/config.toml [hunk 3]
```diff
@@ -4,13 +4,18 @@ root = "/var/lib/containerd-nydus"
...
 daemon_mode = "multiple"
-# Snapshotter's debug and trace HTTP server interface
-enable_system_controller = true
 # Whether tp enable stargz support
 enable_stargz = false
 # Whether snapshotter should try to clean up resources when it is closed
 cleanup_on_close = false
 
+[system]
+# Snapshotter's debug and trace HTTP server interface
```
Comment (imeoer) [inline misc/snapshotter/config.toml:13]: Can the interface address be configurated? Should be documented here.

Comment (changweige) [reply misc/snapshotter/config.toml:13]: Good point. This PR now includes changes that add the config entry to customize the system controller address path

File: misc/snapshotter/config.toml (hunk 1 above)
Comment (changweige) [reply misc/snapshotter/config.toml:21]: Added

File: snapshot/snapshot.go (hunk 2 above)
Comment (changweige) [reply snapshot/snapshot.go]: Renamed

**Review by sctb512 — APPROVED** [review]
LGTM

**Review by imeoer — APPROVED** [review]

**Pull request merged**